src/pages/home/report/ReportActionsList.tsx

```diff
@@ -226,6 +226,7 @@ function ReportActionsList({
         }
         previousLastIndex.current = lastActionIndex;
         reportActionSize.current = sortedVisibleReportActions.length;
+        hasNewestReportActionRef.current = hasNewestReportAction;
```

Comment:
@aldo-expensify We update `hasNewestReportActionRef` whenever `hasNewestReportAction` is changed in this `useEffect` so I think it will not out of sync, as you see we also update other ref like `previousLastIndex`, `reportActionSize` in this `useEffect`.

Reply:
They are updated in different parts of the lifecycle:
- `hasNewestReportAction` changes during rendering
- `hasNewestReportActionRef` changes when evaluating `useEffects`

I'm not familiar enough with the react architecture to be sure that `scrollToBottomForCurrentUserAction` will never be called in between `rendering` and `useEffects` when listening to a "new action event" (I haven't looked at the implementation in `Report.subscribeToNewActionEvent`).

Are you 100% that javascript won't process any other js code (including event listeners) between those two parts of the react lifecycle (rendering/side effects)?

Even if you are sure of the above, I still find it more messy to update this ref here if it is supposed to be in sync with the `hasNewestReportAction`.

Reply:
I agree with that the window is small, but why take the risk?

> Besides, we're updating the ref in useEffect most places in our App, so I think it's ok.

It may be a bad pattern? 🤷 

> I just don't want to update the ref every time the page re-renders and make codes consistency. Wdyt?

What is the problem of updating during rendering? I'm sure there is not performance issue, and for me it is more visible to update it next to where you defined and initialized the variable you are copying. Putting it a `useEffect` makes the code less readable IMO. There may be reasons to do it in other places, but I don't see one here.

Reply:
Thanks, that also makes sense to me now.

Reply:
@aldo-expensify @dukenv0307 I updated to update the ref whenever re-rendering.

Reply:
Thank you both

Reply:
@nkdengineer @dukenv0307 It looks like the refactor has caused a regression [here](https://github.com/Expensify/App/issues/40724#issuecomment-2130786328). 
I am not sure if I understand why this refactor was needed to fix this issue. 
Can you please help me understand this?